--- avrdude-6.3.orig/ft245r.c
+++ avrdude-6.3/ft245r.c
@@ -645,6 +645,7 @@
 cleanup:
     ftdi_usb_close(handle);
 cleanup_no_usb:
+    handle->usb_ctx = NULL;
     ftdi_deinit (handle);
     free(handle);
     handle = NULL;
@@ -658,6 +659,7 @@
         ftdi_set_bitmode(handle, 0, BITMODE_SYNCBB); // set Synchronous BitBang, all in puts
         ftdi_set_bitmode(handle, 0, BITMODE_RESET); // disable Synchronous BitBang
         ftdi_usb_close(handle);
+        handle->usb_ctx = NULL; // avoid SIGSEGV at libusb/io.c:handle_events()
         ftdi_deinit (handle);
         pthread_cancel(readerthread);
         pthread_join(readerthread, NULL);
