--- avrdude-6.4/ft245r.c
+++ avrdude-6.4-new/ft245r.c
@@ -933,7 +933,7 @@
 cleanup:
     ftdi_usb_close(handle);
 cleanup_no_usb:
-    ftdi_deinit (handle);
+    // ftdi_deinit (handle);
     free(handle);
     handle = NULL;
     return -1;
@@ -946,7 +946,7 @@
         ftdi_set_bitmode(handle, 0, BITMODE_SYNCBB); // set Synchronous BitBang, all in puts
         ftdi_set_bitmode(handle, 0, BITMODE_RESET); // disable Synchronous BitBang
         ftdi_usb_close(handle);
-        ftdi_deinit (handle);
+        // ftdi_deinit (handle);
         free(handle);
         handle = NULL;
     }
